inmoose/cohort_qc/qc_report.py

```diff
@@ -16,373 +16,109 @@
-
-        return doc.getvalue()
-
-    def cohort_summary_html_report(self) -> str:
-        """
-        Generate an HTML report summarizing the cohort, including details about the samples and covariates.
-
-        Returns
-        -------
-        str
-            HTML report as a string.
-        """
-        doc, tag, text = Doc().tagtext()
-
-        # Cohort summary
-        summary = self.cohort_qc.cohort_summary()
-        with tag("h2"):
-            text("Cohort Summary")
-
-        with tag("ul"):
-            for key, value in summary.items():
-                if key == "Samples by Covariate Combination":
-                    with tag("li"):
-                        text(f"{key}:")
-                        with tag(
-                            "table",
-                            border="1",
-                            cellpadding="5",
-                            cellspacing="0",
-                            style="border-collapse: collapse;",
-                        ):
-                            # Header row
-                            with tag("tr"):
-                                for covariate in self.cohort_qc.covariates:
-                                    with tag("th"):
-                                        text(covariate)
-                                with tag("th"):
-                                    text("Number of Samples")
-
-                            # Data rows
-                            for combination in value:
-                                with tag("tr"):
-                                    for covariate in self.cohort_qc.covariates:
-                                        with tag("td"):
-                                            text(combination[covariate])
-                                    with tag("td"):
-                                        text(str(combination["Number of Samples"]))
-                else:
-                    with tag("li"):
-                        text(f"{key}: {value}")
-
-        return doc.getvalue()
-
-    def sample_distribution_html_report(self) -> str:
-        """
-        Generate an HTML report comparing the distribution of samples by dataset before and after correction.
-
-        Returns
-        -------
-        str
-            HTML report as a string.
-        """
-        doc, tag, text = Doc().tagtext()
-        with tag("h2"):
-            text("Correction Effect Metric")
-
-        with tag("p"):
-            text(
-                "This metric quantifies the impact of the batch effect correction on the variability of gene expression data. "
-                "The metric is calculated as follows:"
-            )
+        assoc_matrix_before = (
+            self.cohort_qc.association_matrix_before.to_html()
+            if self.cohort_qc.association_matrix_before is not None
+            else None
+        )
+        assoc_matrix_after = self.cohort_qc.association_matrix.to_html()
```

Comment:
`ruff` reports that this variable is never used.
Is it dead code (then consider removing it), or is there something missing? 🤔